src/components/UserActions/UserActions.ts

```diff
@@ -81,16 +81,20 @@ export class UserActions extends Component {
 
         this.options = ComponentOptions.initComponentOptions(element, UserActions, options);
 
-        (get(this.root, UserProfileModel) as UserProfileModel)
-            .getActions(this.options.userId)
-            .then(actions => (actions.length > 0 ? this.render() : this.renderNoActions()))
-            .catch(() => this.renderNoActions());
+        if (!this.options.userId) {
+            this.disable();
```

Comment:
Because you're using a precondition you can remove the else scope if you add a return; after the `this.disable()`.

This will streamline the code even more imo.

Reply:
Even in a constructor ?

Reply:
I don't see why not, if it's an anonymous return.